src/_langs/en/updates/2015-07-22-access-ble-devices-with-web-bluetooth.markdown

```diff
@@ -0,0 +1,279 @@
+
+Web Bluetooth API is at the time of writing partially implemented in Chrome OS
+M45 behind an experimental flag. Go to `chrome://flags/#enable-web-bluetooth`,
+enable the highlighted flag, restart Chrome and you should be able to
+[scan](#scan-bluetooth-devices) and [connect](#connect-to-a-bluetooth-device)
+to nearby Bluetooth devices and
+[read](#read-a-bluetooth-characteristic)/[write](#write-to-a-bluetooth-characteristic)
+Bluetooth characteristics.
+
+<img src="/web/updates/images/2015-07-22-access-ble-devices-with-web-bluetooth/web-bluetooth-flag.png" alt="Web Bluetooth Flag highlighted in chrome://flags"/>
+
+## Security Requirements
+
+### HTTPS Only
+
+Because the Web Bluetooth API is a powerful new feature added to the Web,
+Google Chrome aims to make it [available only to secure
+origins](https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features).
+This means you'll need to build with TLS in mind.
+
+During development you'll be able to play with Web Bluetooth through
+http://localhost, but to deploy it on a site you'll need to have HTTPS setup
+on your server. I personally enjoy [GitHub Pages](https://pages.github.com) for
+demo purposes.
+To add HTTPS to your server then you'll need to get a TLS certificate and set
+it up. Be sure to check out [Mozilla's SSL config
+generator](https://mozilla.github.io/server-side-tls/ssl-config-generator/) for
+best practices there.
+
+### User Gesture Required
+
+As a security feature, discovering nearby Bluetooth devices with
+`navigator.bluetooth.requestDevice` must be called via a user gesture
+like a touch or mouse click.
+
+## Get into the Code
+
+The Web Bluetooth API relies heavily on JavaScript
+[Promises](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+If you're not familiar with them, check out this great
+[Promises tutorial](http://www.html5rocks.com/en/tutorials/es6/promises/). One
+more thing, `() => {}` are simply ECMAScript 2015 [Arrow
+functions](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+- they have a shorter syntax compared to function expressions and lexically
+bind the `this` value.
+
+### Scan Bluetooth Devices
+
+This version of the Web Bluetooth API specification intends to allow websites,
+running in the Central role, to connect to GATT Remote Servers over either a
+Basic Rate/Enhanced Data Rate (BR/EDR) or BLE connection. It also aims to
+support communication among devices that only implement Bluetooth 4.0 or 4.1.
+
+When a website requests access to nearby BLE devices using
+`navigator.bluetooth.requestDevice`, Google Chrome will prompt user with a
+device chooser where he can pick one device or simply cancel the request. At
+the time of writing though, **the device chooser hasn't been implemented yet.**
+Only the closer device that matches filters will be returned.
```

Comment:
not closer, the first device that matches will be returned. It's hard to know what 'first' is, but probably first discovered by ChromeOS. 'first' seems to remain stable.